Code/Framework/AzQtComponents/AzQtComponents/Components/ToastNotification.h

```diff
@@ -31,7 +31,6 @@ namespace AzQtComponents
     {
         Q_OBJECT
     public:    
-        AZ_CLASS_ALLOCATOR(ToastNotification, AZ::SystemAllocator, 0);
```

Comment:
You mentioned that **_system allocator not configured in environment for libAzQtComponents.so_**
Does that mean all classes for AzQtComponents cannot use the system allocator?
![image](https://user-images.githubusercontent.com/69218254/139698661-1800e37e-cfff-4dba-880f-ffb1268bfb48.png)
A quick grep search we can see several other places do that. @amznestebanpapp maybe you could chime in on this issue?

Reply:
I'm not entirely familiar with how the environment is configured from a shared library. seems to work from the editor but when it tries to call a free it hits a null_ptr in Memory.h where AZ::SystemAllocator is not configured.

Reply:
The "system allocator not configured in environment for libAzQtComponents.so" error points to allocators not being initialized properly or the environment not being attached to that dynamic library. 
Removing this line just hides the issue because is going to use C++ allocator, you could get the problem in any other class that is using the SystemAllocator.

Pinging @AMZN-alexpete that has been working in this tool lately.

Reply:
> The "system allocator not configured in environment for libAzQtComponents.so" error points to allocators not being initialized properly or the environment not being attached to that dynamic library. Removing this line just hides the issue because is going to use C++ allocator, you could get the problem in any other class that is using the SystemAllocator.
> 
> Pinging @AMZN-alexpete that has been working in this tool lately.

Any clue why the environment might not be configured. do we have to export it so it would be visible across modules or is there some configuration problem with building the library. that's where I'm stuck at the moment any clue how static variables are shared across modules. I can call init on Memory.h but that doesn't seems like a good fix to the problem since it also just hides the underlying problem.

Reply:
@amznestebanpapp This is actually the correct change for AzQtComponents.
It is a library of Qt Widgets with no AZ::Module hook to attach/deatch an AZ::Environment to.
Furthermore it is shared library that is linked into the executable at linker time and therefore loads before the executable can create the AZ::Environment and unloads after the AZ::Environment has been destroyed

Therefore it cannot use AZ data structures which relies on the AZ Allocators

Reply:
This looks accurate to me - When I moved this widget here from a gem which does have a module, I did not look to see how this library was setup and if there was a module/environment with the allocator settings available.

The next question is should we address the remaining classes in the library in this PR that also refer to the SystemAllocator or in a separate one.

Reply:
@pollend yeah, this problem seems deeper than just this class. The issue is that `AzQtComponents` is a dynamic library. In dynamic libraries, static variables are unique to that dynamic library. To overcome that, we use "environment variables". After a dynamic library is loaded, we pass an "environment variables pointer" to the dll and this one "attaches" it (it basically sets it to a static variable). From that point, the dll sees the same "environment variables" than other dlls/exes (because the static variable points to it). 
I would expect a call to `AZ::Environment::Attach` somewhere in AzQtComponents, which I dont see.

@lumberyard-employee-dm maybe I am missing something. If this is a library of widgets, it should not be a dynamic library. If it is a dynamic library, it should have attach/detach, otherwise nothing related to environment will ever work properly. Removing an allocator from one class is not going to fix the issue, there are multiple as @amzn-rhhong is showing.
I would argue that the solution to not use AZ data structures or AZ allocators is impractical. Wouldn't make this library a static library a better solution? (which may require to remove all the symbol exporting). Or, to add the attach/detach.

Reply:
> This looks accurate to me - When I moved this widget here from a gem which does have a module, I did not look to see how this library was setup and if there was a module/environment with the allocator settings available.

Ah, gotcha, that explains why this showed up. In that case, I am ok with this change and deferring a wider solution for another change. Will approve the change.
 
> The next question is should we address the remaining classes in the library in this PR that also refer to the SystemAllocator or in a separate one.

I would leave it for another PR so we can discuss further what to do.

Reply:
> > This looks accurate to me - When I moved this widget here from a gem which does have a module, I did not look to see how this library was setup and if there was a module/environment with the allocator settings available.
> 
> Ah, gotcha, that explains why this showed up. In that case, I am ok with this change and deferring a wider solution for another change. Will approve the change. Thanks @pollend, sorry for the confusion.
> 
> > The next question is should we address the remaining classes in the library in this PR that also refer to the SystemAllocator or in a separate one.
> 
> I would leave it for another PR so we can discuss further what to do.

The discussion is good, helps with understanding what needs to be done after this.